samza-sql/src/main/java/org/apache/samza/sql/data/SamzaSqlExecutionContext.java

```diff
@@ -38,7 +40,10 @@
    * The variables that are shared among all cloned instance of {@link SamzaSqlExecutionContext}
    */
   private final SamzaSqlApplicationConfig sqlConfig;
-  private final Map<String, UdfMetadata> udfMetadata;
+
+  // Maps the UDF name to list of all UDF methods associated with the name.
+  // Since we support polymorphism there can multiple udfMetadata associated with the single name.
```

Comment:
How can these different methods be invoked via Udf in a sql statement ? Is there an example or a test ? I see one with overload but can we have different method names and make it work ?

Reply:
There was already a testcase (testEndToEndStreamTableTableJoinWithPrimaryKeys) which covers this, because we are passing both object as we all as an integer to the MyTest. I just added another test targeting this specifically.